scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -214,7 +210,14 @@ class Job(val args: Args) extends FieldConversions with java.io.Serializable {
 
   private def executionContext: Try[ExecutionContext] =
     Config.tryFrom(config).map { conf =>
-      ExecutionContext.newContext(conf)(flowDef, mode)
+      val updatedConfig = if (conf.getRequireOrderedSerialization) {
+        state.asScala.foldLeft(conf) {
```

Comment:
why only add these when we require ordered serialization? We want to get the benefit even when we don't require all of them to be ordered. Some of them can still be a win.